src/document/__tests__/DocumentAnnotator-test.ts

```diff
@@ -75,32 +69,27 @@ describe('DocumentAnnotator', () => {
     });
 
     describe('constructor()', () => {
-        test('should add selectionchange event listener', () => {
-            jest.spyOn(document, 'addEventListener');
-
-            annotator = getAnnotator();
+        test('should create highlightListener is feature is enabled', () => {
+            annotator = getAnnotator({ features: { highlightText: true } });
 
-            expect(document.addEventListener).toHaveBeenCalledWith('selectionchange', annotator.handleSelectionChange);
+            expect(annotator.highlightListener).toBeInstanceOf(HighlightListener);
         });
     });
 
     describe('destroy()', () => {
-        test('should clear timeout and remove event handlers', () => {
-            annotator.selectionChangeTimer = 1;
+        test('should remove event handler ', () => {
+            annotator = getAnnotator({ features: { highlightText: true } });
+
             jest.spyOn(annotator, 'removeListener');
-            jest.spyOn(document, 'removeEventListener');
+            jest.spyOn(annotator.highlightListener as HighlightListener, 'destroy');
```

Comment:
Is there a way to avoid this cast?

Reply:
We can use non-null assertion, which is worse than type cast I think.